### ISSUE TITLE
Compression type Snappy is not linked with the binary

### ISSUE THREAD

**Issue by kps116** [romanz/electrs#118]
Hi,
tried to build the docker image like the given docker example. I got the following error on running:<pre>
Feb 26 14:24:21 rbx systemd[1]: Started Docker electrs service.
Feb 26 14:24:23 rbx2 docker[25736]: Config { log: StdErrLog { verbosity: Trace, quiet: false, timestamp: Millisecond, modules: [], writer: "stderr", color_choice: Auto }, network_type: bitcoin, db_path: "./db/mainnet", daemon_dir: "/home/user/.bitcoin", daemon_rpc_addr: V4(127.0.0.1:8332), cookie: Some("xxxx:xxxx"), electrum_rpc_addr: V4(127.0.0.1:50001), monitoring_addr: V4(127.0.0.1:4224), jsonrpc_import: false, index_batch_size: 100, bulk_index_threads: 4, tx_cache_size: 10000, server_banner: "Welcome to electrs (Electrum Rust Server)!" }
Feb 26 14:24:23 rbx2 docker[25736]: 2019-02-26T13:24:23.597+00:00 - DEBUG - Server listening on 127.0.0.1:4224
Feb 26 14:24:23 rbx2 docker[25736]: 2019-02-26T13:24:23.661+00:00 - DEBUG - Running accept thread
Feb 26 14:24:23 rbx2 docker[25736]: 2019-02-26T13:24:23.661+00:00 - INFO - NetworkInfo { version: 170100, subversion: "/Satoshi:0.17.1/" }
Feb 26 14:24:23 rbx2 docker[25736]: 2019-02-26T13:24:23.662+00:00 - INFO - BlockchainInfo { chain: "main", blocks: 564743, headers: 564743, bestblockhash: "0000000000000000001942e691bd2cfce723f787e878a8e06d9666c259feab18", pruned: false, initialblockdownload: false }
Feb 26 14:24:23 rbx2 docker[25736]: 2019-02-26T13:24:23.662+00:00 - DEBUG - opening DB at "./db/mainnet"
Feb 26 14:24:23 rbx2 docker[25736]: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { message: "Invalid argument: Compression type Snappy is not linked with the binary." }', src/libcore/result.rs:1009:5
Feb 26 14:24:23 rbx2 docker[25736]: note: Run with `RUST_BACKTRACE=1` for a backtrace.
Feb 26 14:24:24 rbx2 systemd[1]: electrs.service: main process exited, code=exited, status=101/n/a
</pre>
Since I am not familiar with rust or rocksdb and the build process, do you see any hint what to do?
Thanks, Klaus

**Comment by pruflyos** [comment]
I ran into the same issue using Alpine Linux as the base image for the Docker container.

Maybe these links are helpful in finding a solution:
https://stackoverflow.com/questions/8808691/can-i-build-a-shared-library-by-linking-static-libraries
https://github.com/unoexperto/docker-rocksdb/blob/master/Dockerfile

**Comment by pruflyos** [comment]
Seems like the pure presence of `snappy-dev` during compile time is not enough, the resulting `electrs` binary still throws the error reported by @kps116 

My Dockerfile contains :
```
FROM alpine:latest
RUN apk --no-cache add --update alpine-sdk
RUN apk --no-cache add --update clang
RUN apk --no-cache add --update linux-headers
RUN apk --no-cache add --update cargo
RUN apk --no-cache add --update zlib-dev bzip2-dev snappy-dev lz4-dev zstd-dev
...
RUN cargo install electrs --version 0.4.3
```

**Comment by romanz** [comment]
Thanks for reporting this issue!
Please try the latest Dockerfile (from https://github.com/romanz/electrs/commit/bb8baf410839f2ff1e1996b705b5fa7ec93b8a5f) - it should (hopefully) build correctly.

**Comment by pruflyos** [comment]
TLDR: `electrs 0.5.0` seems to fix this

I didn't try with your Dockerfile, because the resulting image is huge (500+ MB). With our Alpine Linux based Dockerfile we end up with a 8MB Dockerfile. However, we applied the switch to `electrs 0.5.0` to our Dockerfile as well and the issue regarding the missing snappy linking in the binary is fixed.
Although I don't really understand what change between `0.4.3` and `0.5.0` is exactly responsible for the fix.
